addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java

```diff
@@ -376,6 +253,41 @@ public void packetReceived(byte[] packet) {
                     if (msg.subType == SubType.RESPONSE) {
                         logger.debug("RFXCOM transceiver/receiver type: {}, hw version: {}.{}, fw version: {}",
                                 msg.transceiverType, msg.hardwareVersion1, msg.hardwareVersion2, msg.firmwareVersion);
+
+                        if (msg.command == Commands.GET_STATUS) {
+                            if (configuration.ignoreConfig) {
+                                logger.debug("Ignoring transceiver configuration");
+                            } else {
+                                byte[] setMode = null;
+
+                                if (configuration.setMode != null && !configuration.setMode.isEmpty()) {
+                                    try {
+                                        setMode = DatatypeConverter.parseHexBinary(configuration.setMode);
+                                        if (setMode.length != 14) {
+                                            logger.warn("Invalid RFXCOM transceiver mode configuration");
+                                            setMode = null;
+                                        }
+                                    } catch (IllegalArgumentException ee) {
+                                        logger.warn("Failed to parse setMode data", ee);
+                                    }
+                                } else {
+                                    setMode = RFXComMessageFactory.setMode(msg.transceiverType, configuration);
```

Comment:
If you move the method there the name setMode (I would expect a setter of the mode field) might be unclear, we actually ask the factory to generate the setMode message. `getSetModeMessage`, `generateSetModeMessage`.

But since this an outgoing message, I would prefer to keep using the message class, I see no direction reason why we should differentiate on using message classes for some but not for all. *And now you mention it* I see the other hard-coded message in the *RFXComMessageFactory*, however there is a difference between hard coded and dynamically constructed message. 

I think that I would prefer to keep the message class and that we could have a setter to pass in the complete configuration, do you also see that as a viable option?